storage-proofs/porep/src/stacked/vanilla/cores.rs

```diff
@@ -0,0 +1,189 @@
+use log::*;
+use std::sync::{Arc, Mutex, MutexGuard};
+
+use anyhow::Result;
+use hwloc::{ObjectType, Topology, TopologyObject, CPUBIND_THREAD};
+use lazy_static::lazy_static;
+
+use storage_proofs_core::settings;
+
+type CoreGroup = Vec<CoreIndex>;
+lazy_static! {
+    pub static ref TOPOLOGY: Arc<Mutex<Topology>> = Arc::new(Mutex::new(Topology::new()));
```

Comment:
why is this `Arc`, `Mutex`? it seems to me this is never changed, so just `Topology` should be enough

Reply:
The `set_cpubind_for_thread` call requires it to be mutable. I can get rid of the `Arc`, though. It was needed at one point along the way but not was things ended up.